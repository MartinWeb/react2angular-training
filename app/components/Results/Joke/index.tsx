import { react2angular } from 'react2angular';
import { Joke, JokeProps } from './Joke';

const JokeComponent = react2angular((props: JokeProps) => <Joke {...props} />, ['joke']);

export { JokeComponent, Joke };
