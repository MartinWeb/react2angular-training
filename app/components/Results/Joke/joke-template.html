<pre><h3>{{$ctrl.joke.value}}</h3></pre>