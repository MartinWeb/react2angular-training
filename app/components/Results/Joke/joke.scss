pre {
    max-width: 80%;
    white-space: pre-wrap;
    margin: 10px auto;
    border: 1px solid black;
    padding: 10px;
    background: cornsilk;
}