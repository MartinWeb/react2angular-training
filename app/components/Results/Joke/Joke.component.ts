import angular from "angular";
import template from './joke-template.html';
import './joke.scss';

export const JokeComponent: angular.IComponentOptions = {
    template,
    bindings: {
        joke: '='
    }
};