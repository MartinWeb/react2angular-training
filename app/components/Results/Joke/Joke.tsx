import './joke.scss';

export type JokeProps = {
    joke: ChuckNorris.Joke;
}

export const Joke = ({ joke }: JokeProps) => (
    <pre><h3>{joke?.value}</h3></pre>
);
