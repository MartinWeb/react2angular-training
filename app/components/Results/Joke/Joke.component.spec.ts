import { JokeComponent } from "./Joke.component";

describe('JokeComponent', () => {
    test('should have valid snapshot', () => {
        expect(JokeComponent.template).toMatchSnapshot();
    });

    test('should have valid bindings', () => {
        expect(JokeComponent.bindings).toMatchSnapshot();
    });
});