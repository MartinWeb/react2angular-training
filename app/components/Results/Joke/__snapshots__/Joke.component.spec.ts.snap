// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JokeComponent should have valid bindings 1`] = `
{
  "joke": "=",
}
`;

exports[`JokeComponent should have valid snapshot 1`] = `"<pre><h3>{{$ctrl.joke.value}}</h3></pre>"`;
